tree/ntuple/v7/inc/ROOT/RFieldVisitor.hxx

```diff
@@ -126,19 +140,51 @@ public:
       fAvailableSpaceValueString = fWidth - 6 - fAvailableSpaceKeyString;
    }
 };
-
+// clang-format off
+/**
+\class ROOT::Experimental::RValueVisitor
+\ingroup NTuple
+\brief Traverses through an ntuple to display its entries.
+    
+Each visit outputs the entry of a single field as in a .json file.
+*/
+// clang-format on
+// NOTE (lesimon): currently no support for char, std::array, standard arrays,
+//    RField<Clustersize_t>, not sure about std::int64_t.
+class RValueVisitor: public Detail::RNTupleVisitor {
+private:
+   RNTupleReader* fReader;
+   std::ostream &fOutput;
+   int fIndex;
+public:
+   RValueVisitor(std::ostream &output, RNTupleReader* reader, int index): fReader{reader}, fOutput{output}, fIndex{index} {}
+   void VisitField(const Detail::RFieldBase &field, int level) final;
+   void VisitRootField(const RFieldRoot &/*fField*/, int /*level*/) final { }
+   void VisitArrayField(const RFieldArray &field, int level, std::size_t arraylength, const Detail::RFieldBase* /*subfieldPtr*/) final;
```

Comment:
The name does not need to be commented, i.e. write `const Detail::RFieldBase *subField`